components/molecules/ContributorListTableRow/contributor-list-table-row.tsx

```diff
@@ -48,7 +48,14 @@ function getTopContributorLanguages(contributor: DbUser) {
   // some contributors will have empty language objects so we will pull their popular language from the interests field instead of defaulting to nothing
   const entries = Object.entries<string>(contributor.languages);
 
-  return (entries.length === 0 ? contributor.interests.split(",") : entries)
+  if (entries.length === 0) {
```

Comment:
I’m on mobile at the moment, but is the only difference that we trim interests once they’re split now? I’m not sure how the trim changes much. Also, the new code returns all interests instead of two.

Reply:
the trim is just me being over cautious 😂. but the main change here is having interest sorted and sliced seperately from the entries since it ends up being an `[string, string][]` type

Reply:
Not sure why that is because it should return one array or the other in the old code.

I'm still not sure why (condition ? array1 : array2) resulted in a tuple `[string, string]` though.

Anyways, mind testing to be certain all is good when there are no most used languages and interests are picked up instead?

Reply:
I see what the issue is with the `[string, string]`. It's because of getting the object entries.

We can do this instead.

```typescript
function getTopContributorLanguages(contributor: DbUser) {
  // some contributors will have empty language objects so we will pull their popular language from the interests field instead of defaulting to nothing
  const entries = Object.entries<string>(contributor.languages);
  const languages = entries.length > 0 ? entries.map(([language]) => language) : contributor.interests.split(",");

  return languages.sort((a, b) => (a < b ? -1 : 1)).slice(0, 2);
}
```

I'm going to make that change and also add a unit test.